Additional null check for Option provided to factory methods generated when @Value.Parameter is used.

MFAshby opened the pull request:
Additional fix and tests for https://github.com/immutables/immutables-vavr/issues/27

MFAshby committed:
Additional null check for Option provided to factory methods generated when @Value.Parameter is used.

https://github.com/immutables/immutables-vavr/issues/27

elucash commented on vavr-encodings/src/main/java/org/immutables/vavr/encodings/VavrOptionEncoding.java:
```diff
@@ -47,6 +47,12 @@
 
+  @Encoding.Of
+  static <T> Option<T> init(final Option<T> value) {
+    Objects.requireNonNull(value);
+    return value;
```
```suggestion
    return Objects.requireNonNull(value);
```

MFAshby committed:
Update vavr-encodings/src/main/java/org/immutables/vavr/encodings/VavrOptionEncoding.java

`Objects.requireNonNull(value)` actually also returns the value it was given after asserting non-nullness.

Co-authored-by: Eugene Lukash <e.lucash@gmail.com>

io7m reviewed (approved):
Looks good!
